pkg/config/constants/constants.go

```diff
@@ -59,6 +59,9 @@ const (
 	// IstioIngressNamespace is the namespace where Istio ingress controller is deployed
 	IstioIngressNamespace = "istio-system"
 
+	// DefaultKubernetesDomain the default service domain suffix for Kubernetes, if not overridden in config.
+	DefaultKubernetesDomain = "cluster.local"
```

Comment:
Can we simplify this ? Since it's not really a constant, but a setting - just make sure all
code that is using cluster.local gets a reference to the MeshConfig or ProxyConfig object - where this should be defined ? Unfortunately we're still using a mix of flags/env - but we
are trying to have all config in that affect proxies ProxyConfig and the rest in MeshConfig

Reply:
Unfortunately, it *is* a constant currently ... a lot of code just uses it as a constant. It's way beyond of the scope of this PR to fix it, but we definitely should. #23652